Demisto-sdk release 1.33.1 (#4737)
demisto/demisto-sdk: CHANGELOG.md

# Changelog
## 1.33.1 (2024-12-25)
### Feature

> * Moved the CR101 validation, checking names of Correlation Rule files match standards, to `validate_content_path`. [#4596](https://github.com/demisto/demisto-sdk/pull/4569)

Reviewer: ```suggestion
* Moved the CR101 validation to the new validation format. Checks names if "Correlation Rule" files match standards to `validate_content_path`. [#4596](https://github.com/demisto/demisto-sdk/pull/4569)
```

> * Added validation checks whether the silent playbook name id and the isSilent key are set correctly. [#4662](https://github.com/demisto/demisto-sdk/pull/4662)

Reviewer: ```suggestion
* Added *PB130* validation. Checks whether the silent playbook name id and the *isSilent* key are set correctly. [#4662](https://github.com/demisto/demisto-sdk/pull/4662) PB130
```

> * Added *PB130* validation. Checks whether the silent playbook name id and the *isSilent* key are set correctly. [#4662](https://github.com/demisto/demisto-sdk/pull/4662) PB130

Reviewer: ```suggestion
* Added *PB130* validation. Checks whether the silent playbook name id and the *isSilent* key are set correctly. [#4662](https://github.com/demisto/demisto-sdk/pull/4662)
```

> * Added validation to ensure every silent trigger points to a silent playbook, and vice versa. [#4670](https://github.com/demisto/demisto-sdk/pull/4670)

Reviewer: ```suggestion
* Added *PB131* validation. Ensures every silent trigger points to a silent playbook, and vice versa. [#4670](https://github.com/demisto/demisto-sdk/pull/4670)
```

* Exclude silent-Playbooks/Triggers from the metadata. [#4692](https://github.com/demisto/demisto-sdk/pull/4692)

> * Added support for writing pre-commit results to files. [#4697](https://github.com/demisto/demisto-sdk/pull/4697)

Reviewer: ```suggestion
* Added support for writing ***demisto-sdk pre-commit*** command results to files. [#4697](https://github.com/demisto/demisto-sdk/pull/4697)
```

> * Added support for CaseLayouts and CaseFields content items paths to the `validate-content-paths` **pre-commit** hook. [#4706](https://github.com/demisto/demisto-sdk/pull/4706)

Reviewer: ```suggestion
* Added support for CaseLayouts and CaseFields content items paths to the `validate-content-paths` ***demisto-sdk pre-commit*** hook. [#4706](https://github.com/demisto/demisto-sdk/pull/4706)
```

* Updated validation rules to allow the deletion of silent Playbooks and silent Triggers. [#4715](https://github.com/demisto/demisto-sdk/pull/4715)
* Exclude silent items from release notes validation. [#4720](https://github.com/demisto/demisto-sdk/pull/4720)

> * Added the **PB132** validation to ensure that silent playbooks do not have a README file. [#4723](https://github.com/demisto/demisto-sdk/pull/4723)

Reviewer: ```suggestion
* Added *PB132* validation. Ensures that silent playbooks do not have a README file. [#4723](https://github.com/demisto/demisto-sdk/pull/4723)
```

* Added support for Silent-Playbooks in the old-validate. [#4726](https://github.com/demisto/demisto-sdk/pull/4726)

### Fix
* Fixed an issue where YmlSplitter attributes were being unintentionally updated. [#4696](https://github.com/demisto/demisto-sdk/pull/4696)

> * Fix an issue where *RN106* and *PA114* validations would fail on new packs. [#4710](https://github.com/demisto/demisto-sdk/pull/4710)

Reviewer: ```suggestion
* Fixed an issue where *RN106* and *PA114* validations would fail on new packs. [#4710](https://github.com/demisto/demisto-sdk/pull/4710)
```

* Fixed an issue where *RN107* validation would fail on new content items and specific content types. [#4711](https://github.com/demisto/demisto-sdk/pull/4711)
* Fixed an issue where YmlSplitter attributes were being unintentionally updated. [#4713](https://github.com/demisto/demisto-sdk/pull/4713)

> * Fixed an issue where RN111 would fail when it should not when the docker entry message was missing. [#4714](https://github.com/demisto/demisto-sdk/pull/4714)

Reviewer: ```suggestion
* Fixed an issue where *RN111* would fail when it should not when the docker entry message was missing. [#4714](https://github.com/demisto/demisto-sdk/pull/4714)
```

> * Fixed an issue where ST110 would incorrectly fail when adding the isSilent field. [#4716](https://github.com/demisto/demisto-sdk/pull/4716)

Reviewer: ```suggestion
* Fixed an issue where *ST110* would incorrectly fail when adding the isSilent field. [#4716](https://github.com/demisto/demisto-sdk/pull/4716)
```

> * Fixed an issue where RM114 falsely failed when it concatenated "Packs/" twice to the file path. [#4717](https://github.com/demisto/demisto-sdk/pull/4717)

Reviewer: ```suggestion
* Fixed an issue where *RM114* falsely failed when it concatenated "Packs/" twice to the file path. [#4717](https://github.com/demisto/demisto-sdk/pull/4717)
```

* Fixed an issue where the error message wasn't clear when attempting to upload a content item to an unsupported marketplace. [#4727](https://github.com/demisto/demisto-sdk/pull/4727)
* Changed the isSilent key to be lower case. [#4728](https://github.com/demisto/demisto-sdk/pull/4728)

> * Fixed an issue in the ***modeling-rules*** command where colored logs were not printed properly. [#4733](https://github.com/demisto/demisto-sdk/pull/4733)

Reviewer: ```suggestion
* Fixed an issue in the ***demisto-sdk modeling-rules*** command where colored logs were not printed properly. [#4733](https://github.com/demisto/demisto-sdk/pull/4733)
```

### Breaking

> * Removed support for DS107 and RM106 validations - ensure no 'demist'o word in description and readme files for both new & old validate. [#4712](https://github.com/demisto/demisto-sdk/pull/4712)

Reviewer: ```suggestion
* Removed support for *DS107* and *RM106* validations. Ensures no "demisto" word in description and readme files for both new & old validate formats. [#4712](https://github.com/demisto/demisto-sdk/pull/4712)
```

> * Changed the code of RM116 - Validate that the readme file is not to short to RM117 due to error code duplication. [#4712](https://github.com/demisto/demisto-sdk/pull/4712)

Reviewer: ```suggestion
* Changed the code of *RM116*. Validates that the readme file is not to short to *RM117* due to error code duplication. [#4712](https://github.com/demisto/demisto-sdk/pull/4712)
```
